File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {

        int n = nums.length;
        int left = 0;
        int right = n - 1;

        while(left<=right){
            int mid = left - (left-right)/2;

            // if mid element is the target then return mid
            if(nums[mid] == target){
                return mid;
            }

            if(nums[mid] < target){
                left = mid + 1;
            }else{
                right = mid - 1;
            }
        }

        return left;
        
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:

        n = len(nums)
        left, right = 0, n-1

        while left<=right:

            mid = left - (left -right)//2

            if nums[mid] == target:
                return mid
            
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1

        return left
        